File: app/models/Recipe.scala
package models;

import javax.persistence._
import scala.beans.BeanProperty

@Entity
@Table(name="recipe")
class recipe(recipeName : String){
  
  @Id
  @GeneratedValue(strategy = GenerationType.SEQUENCE)
  @Column(name="id", nullable= false)
  @BeanProperty
  var id: Int = _

  @Column(name="name", nullable=false, length=100)
  @BeanProperty
  var name: String = recipeName
  
  @Column(name="description")
  @BeanProperty
  var description: String = _

  @Column(name="owner")
  @BeanProperty
  var owner: String = _   
  
 // @OneToMany(cascade = CascadeType.ALL)
 // var instructions : Instruction = _

}

File: app/models/Recipe.java
package models;

import javax.persistence.*;

import com.avaje.ebean.Model;
import entities.CoursesEnum;

/**
 * Recipe entity managed by Ebean
 */
@Entity
@Table(name = "recipe")
public class Recipe extends Model{
  
  @Id @GeneratedValue
  //@Column(name="id", nullable= false)
  public Long id;
  
  @Column(name="name", nullable= false)
  public String name;
  
  @Column(name="description")
  public String description;

  @Column(name="owner", length=255)
  private String owner;
  
  @Column(name="course")
  private CoursesEnum course;
  
  @OneToMany
  public Instruction instructions;
  
}